packages/nc-gui-v2/pages/index/index/create-external.vue

```diff
@@ -77,51 +93,91 @@ const validators = computed(() => {
 
 const { validate, validateInfos } = useForm(formState, validators)
 
+const populateName = (v: string) => {
+  formState.dataSource.connection.database = `${v.trim()}_noco`
+}
+
 const onClientChange = () => {
   formState.dataSource = { ...getDefaultConnectionConfig(formState.dataSource.client) }
+  populateName(formState.title)
 }
 
-const inflectionTypes = ['camelize', 'none']
-const configEditDlg = ref(false)
+const onSSLModeChange = ((mode: 'No' | 'Allow' | string) => {
+  switch (mode) {
```

Comment:
`if ('ssl' in formState.dataSource.connection)`
otherwise TS complains that `.ssl` does not exist on `formState.dataSource.connection`

Reply:
Yes but this actually breaks onSSLModeChange, as it manipulates ssl property by creating or removing, so it had to run in both cases

Reply:
Ah then our structure is somewhat bad / type unsafe 
For now you can just type cast it to `Record<string, any>` I guess 😅